crates/bevy_reflect/src/reflect.rs

```diff
@@ -149,7 +165,12 @@ pub trait Reflect: Any + Send + Sync {
     /// containing the trait object.
     fn set(&mut self, value: Box<dyn Reflect>) -> Result<(), Box<dyn Reflect>>;
 
-    /// Returns an enumeration of "kinds" of type.
+    /// Returns a simple enumeration of "kinds" of type, without any associated object.
```

Comment:
Nit: This comment should be specific to "the value", rather than all possible values. For example, possibly something like:

```suggestion
    /// Returns the value's [kind](ReflectKind).
```

I know this is based on the docs from `reflect_ref` and friends, but imo those also need to be updated (not necessarily in this PR, though).